### PR TITLE
convert from govendor to dep

### PR DESCRIPTION
**Pull request by bryanl**
Convert from govendor to dep. 

_note_: this updates a few packages as well.

* [x] convert to dep
* [x] updating contrib docs

Signed-off-by: bryanl <bryanliles@gmail.com>

### CODE REVIEW

**Comment by hausdorff** [comment]
I think it's worth moving to `dep`, but it's hard for me to tell how confident we are in these changes.

1. Does this PR complete the transition to `dep`? (I don't see a message like "Fixes #197".) If not, what remains?
1. Did we resolve the problems Gus had when he tried to transition over ksonnet/kubecfg (as described in #197)?

**Comment by bryanl** [comment]
@hausdorff those issues haven't presented themselves in this change. Also, we can potentially update `client-go` to the latest release (6.x) and that should solve the dep/client-go issues raised in #197.

**Comment by bryanl** [comment]
@hausdorff Also, yes this completes the transition to `dep`.

**Comment by hausdorff** [comment]
@bryanl We should upgrade `client-go`, but I think we should do it in a different review. It's a pretty big operational risk (1) because we have accumulated a lot of _ad hoc_ fixes for bugs in the current version, and (2) they do break the internal APIs all the time.

I'll take a look at the review in a bit.

**Comment by hausdorff** [comment]
@bryanl Oh, also, if we're going to transition to `dep` and upgrade `client-go`, we should amortize that risk by making sure we upgrade `kubecfg` on both those things at the same time. Bitnami has to use that code in anger, and if there are issues, they will surface quite a bit faster when vetted through there.

**Comment by bryanl** [comment]
The goal was to make the changes in steps. 1) move to dep 2) update client-go and friends. This PR Is only the first step.

**Comment by hausdorff** [comment]
@bryanl ok, so, just to be clear: the plan _is_ to update kubecfg as well, yes?

**Comment by hausdorff** [comment]
Summarizing offline conversations: yes, we're planning on updating kubecfg.

**Comment by bryanl** [comment]
@hausdorff I'm going to review to see if that is possible. dep and govendor. I may have to go back and adjust each dep by hand. The problem is explicit vs implicit dependencies.

**Comment by hausdorff** [comment]
@bryanl Ok, if it's not easy, then I recommend just shipping it as-is. Another way to ameliorate this risk is to ship a release candidate for 0.9, so that at least people have a chance to test it in the wild. (I bring this up because we have a bunch of version-specific fixes for bugs in our dependencies, strung throughout the code, so. Ick.)

**Comment by jessicayuen** [comment]
Just incase you miss this, should do a rebase before merging. Some commits are dirty.

**Comment by jessicayuen** [comment]
I've noticed this is taking a lot longer to build (even on subsequent builds). Are you seeing similar behavior @bryanl ?

```
➜  ksonnet git:(pr-265) ✗ time make
go build -o ks -ldflags="-X main.version=foo "  .
./docs/generate/update-generated-docs.sh
make  42.67s user 4.83s system 209% cpu 22.687 total
```

**Comment by bryanl** [comment]
I didn't notice. Will investigate in the morning.

File: Gopkg.lock [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+  revision = "21300e3e11c918b8e6a70fb7293b310683d6c046"
+  version = "v3.0.0"
```
Comment (redbaron) [inline Gopkg.lock]: this  is for 1.6 Kubernetes, does it mean that more recent features like CRDs are not supported?

**Pull request merged**